controller/eks-cluster-config-handler.go

```diff
@@ -883,26 +885,101 @@ func (h *Handler) updateUpstreamClusterState(upstreamSpec *v13.EKSClusterConfigS
 		desiredNgVersions[ng.NodegroupName] = desiredVersion
 	}
 
-	var attemptUpgradingNodegroups bool
-	for _, ng := range upstreamSpec.NodeGroups {
-		if aws.StringValue(ng.Version) == desiredNgVersions[ng.NodegroupName] {
+	var attemptUpdatingNodegroups bool
+	for _, upstreamNg := range upstreamSpec.NodeGroups {
+		// if continue is used after an update, it means that update
+		// must finish before others for that nodegroup can take place.
+		// Some updates such as minSize, maxSize, and desiredSize can
+		// happen together
+
+		ng := ngs[upstreamNg.NodegroupName]
```

Comment:
do we need to check if this upstream node group exists in the map?

Reply:
No because above this is logic to create and then return if the nodegroup is not in the upstream map. This point will not be reached if that's the case.